firedrake/ensemble.py

```diff
@@ -21,37 +23,28 @@ def __init__(self, comm, M):
 
         rank = comm.rank
 
-        # User global comm
+        # Global comm
         self.global_comm = comm
         # Internal global comm
-        self._global_comm = internal_comm(comm, self)
+        self._comm = internal_comm(comm, self)
 
-        # User split comm
+        # User and internal communicator for spatial parallelism, contains a
+        # contiguous chunk of M processes from `global_comm`.
         self.comm = self.global_comm.Split(color=(rank // M), key=rank)
-        # Internal split comm
-        self._comm = internal_comm(self.comm, self)
-        """The communicator for spatial parallelism, contains a
-        contiguous chunk of M processes from :attr:`global_comm`"""
+        self.comm.name = "Ensemble spatial comm"
```

Comment:
If these names are used for debugging could we have an optional str argument for an Ensemble name? So these would be
```
comm_name = "Ensemble" if optional_comm_name is None else optional_comm_name
self.comm.name = f"{comm_name} spatial comm"
```